nautobot/extras/api/views.py

```diff
@@ -289,6 +287,195 @@ class ApprovalWorkflowViewSet(NautobotModelViewSet):
+
+            if not self._has_change_permission(user, workflow):
+                ct = workflow.object_under_review_content_type
+                model_name = ct.model
+                app_label = ct.app_label
+                return Response(
+                    {"detail": f"You do not have 'change' permission on {app_label}.{model_name}."},
+                    status=status.HTTP_403_FORBIDDEN,
+                )
+
+            if not self._is_user_approver(user, stage):
+                return Response(
+                    {"detail": "You do not have permission to approve this stage."},
+                    status=status.HTTP_403_FORBIDDEN,
+                )
+
+            if self._user_already_responded(user, stage):
+                return Response(
+                    {"detail": "You have already responded to this stage."},
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+
+            comment = request.data.get("comments", "")
+            ApprovalWorkflowStageResponse.objects.create(
+                approval_workflow_stage=stage,
+                user=user,
+                state=ApprovalWorkflowStateChoices.APPROVED
+                if action_type == "approve"
+                else ApprovalWorkflowStateChoices.DENIED,
+                comments=comment,
+            )
+            serializer = serializers.ApprovalWorkflowSerializer(workflow, context={"request": request})
+            return Response(serializer.data, status=status.HTTP_200_OK)
+
+        except ApprovalWorkflow.DoesNotExist:
+            return Response(
+                {"detail": "ApprovalWorkflow not found."},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+
+    def restrict_queryset(self, request, *args, **kwargs):
+        """
+        Apply special permissions as queryset filter on the /approve/, /deny/, and /comment/ endpoints.
+
+        Otherwise, same as ModelViewSetMixin.
+        """
+        action_to_method = {"approve": "change", "deny": "change", "comment": "change"}
+        if request.user.is_authenticated and self.action in action_to_method:
+            self.queryset = self.queryset.restrict(request.user, action_to_method[self.action])
+        else:
+            super().restrict_queryset(request, *args, **kwargs)
+
+    class ApprovalWorkflowStageChangePermission(TokenPermissions):
+        """Enforce `change_approvalworkflowstage` permission (instead of default `add_approvalworkflow` for POST)."""
+
+        perms_map = {
+            "POST": ["%(app_label)s.change_%(model_name)s"],
+        }
+
+    @extend_schema(
+        methods=["post"],
+        request=None,
+        responses={"200": serializers.ApprovalWorkflowSerializer},
+    )
+    @action(
+        detail=True,
+        methods=["post"],
+        permission_classes=[ApprovalWorkflowStageChangePermission],
+    )
+    def approve(self, request, pk=None):
+        """Current user approves the active stage of this workflow."""
```

Comment:
IIRC these docstrings appear in Swagger docs for these endpoints, so perhaps something less passive-voice, like `"Approve the active stage of this workflow."